### PR TITLE
WDPD-194 Frontend handling Payolution B2B

### PR DESCRIPTION
**Pull request by ps-cgrach**
### This PR

Add front end handling of Payolution B2B

- show mandatory company name field if not already provided
- don't show the date of birth field
- check's to show payment method

### How to test

Activate Payolution B2B and check for proper frontend handling

### Jira Ticket
https://jira.parkside.at/browse/WDPD-194

### CODE REVIEW

File: views/blocks/order_button.tpl [hunk 1]
```diff
@@ -2,62 +2,64 @@
...
+  [{$smarty.block.parent}]
+  [{/if}]
 
 [{include file="sepa_mandate_modal.tpl"}]
 
-[{if $payment->oxpayments__oxid->value == "wdpayolution-inv" && $payment->oxpayments__terms->value}]
-    <script>
-        var checkbox = document.getElementById("terms-checkbox");
-        var button = document.getElementById("payolution-inv-order-button");
+[{if $isConsentNeeded}]
```
Comment (ps-amehic) [inline views/blocks/order_button.tpl]: A question: is the same logic with had for this checkbox for Payolution Inv now just expanded to PayolutionB2B as well?

Comment (ps-cgrach) [reply views/blocks/order_button.tpl]: Yes, they are the same. One is for regular customers and one is for business customers.

File: views/blocks/order_button.tpl [hunk 2]
```diff
@@ -2,62 +2,64 @@
...
+
+    <div class="hidden">
+      [{$oViewConf->getHiddenSid()}]
+      [{$oViewConf->getNavFormParams()}]
+      <input type="hidden" name="cl" value="order">
+      <input type="hidden" name="fnc" value="[{$oView->getExecuteFnc()}]">
+      <input type="hidden" name="challenge" value="[{$challenge}]">
+      <input type="hidden" name="sDeliveryAddressMD5" value="[{$oView->getDeliveryAddressMD5()}]">
+
+      [{if $oView->isActive('PsLogin') || !$oView->isConfirmAGBActive()}]
```
[the remark this answers is not in the document]
Comment (ps-cgrach) [reply views/blocks/order_button.tpl]: 😄 done!

File: Model/PayolutionBtwobPaymentMethod.php [hunk 3]
```diff
@@ -97,9 +109,59 @@ public function addMandatoryTransactionData(&$oTransaction, $oOrder)
...
+    }
+
+    /**
+     * Checks if the user is older than 18 or the date of birth needs to be entered
+     *
+     * @return bool
+     *
+     * @throws \Exception
+     *
+     * @since 1.2.0
```
[the remark this answers is not in the document]
Comment (ps-cgrach) [reply Model/PayolutionBtwobPaymentMethod.php]: 👍

File: Extend/Model/PaymentGateway.php [hunk 4]
```diff
@@ -10,23 +10,19 @@
 namespace Wirecard\Oxid\Extend\Model;
 
 use Exception;
-
```
[the remark this answers is not in the document]
Comment (ps-cgrach) [reply Extend/Model/PaymentGateway.php]: 👍

File: Model/PaymentMethod.php [hunk 5]
```diff
@@ -352,4 +352,16 @@ public function isPaymentPossible()
...
     }
+
+    /**
+     * Get the keys that should not be included for this payment method
+     *
+     * @return array
+     *
+     * @since 1.3.0
+     */
+    public function hiddenAccountHolderFields()
```
[the remark this answers is not in the document]
Comment (ps-cgrach) [reply Model/PaymentMethod.php]: 👍

File: Model/PayolutionBtwobPaymentMethod.php [hunk 6]
```diff
@@ -94,9 +109,59 @@ public function addMandatoryTransactionData(&$oTransaction, $oOrder)
...
+    /**
+     * Checks if the user is older than 18 or the date of birth needs to be entered
+     *
+     * @return bool
+     *
+     * @throws \Exception
+     *
+     * @since 1.3.0
+     */
+    protected function _checkDateOfBirth()
```
Comment (tomazpu) [inline Model/PayolutionBtwobPaymentMethod.php]: For me it makes not much sense to check in an B to B payment method for the birth date? do you agree? We need to clear it with @hbrandl

Comment (ps-dfritzsch) [reply Model/PayolutionBtwobPaymentMethod.php]: We personally agree to this as well, but it was already discussed with Hartwig via phone.

Comment (ps-dfritzsch) [reply Model/PayolutionBtwobPaymentMethod.php]: We agree to check it if the birthdate is available in the user account (not mandatory to enter a birthdate).

**Pull request merged**